packages/fast-components-react-msft/src/call-to-action/call-to-action.spec.tsx

```diff
@@ -0,0 +1,172 @@
+import * as React from "react";
+import * as ShallowRenderer from "react-test-renderer/shallow";
+import * as Adapter from "enzyme-adapter-react-16/build";
+import { configure, shallow } from "enzyme";
+import examples from "./examples.data";
+import { generateSnapshots } from "@microsoft/fast-jest-snapshots-react";
+import {
+    CallToActionAppearance,
+    CallToActionProps,
+    ICallToActionHandledProps,
+    ICallToActionManagedClasses,
+    ICallToActionUnhandledProps
+} from "./call-to-action";
+import { ICallToActionClassNameContract } from "@microsoft/fast-components-class-name-contracts-msft";
+
+/*
+ * Configure Enzyme
+ */
+configure({adapter: new Adapter()});
+
+describe("call to action snapshot", (): void => {
+    generateSnapshots(examples);
+});
+
+describe("call to action", (): void => {
+    const Component: React.ComponentClass<ICallToActionHandledProps & ICallToActionManagedClasses> = examples.component;
+    const managedClasses: ICallToActionClassNameContract = {
+        callToAction: "call-to-action",
+        callToAction_glyph: "glyph"
+        callToAction__primary: "call-to-action-primary",
+        callToAction__lightweight: "call-to-action-lightweight",
+        callToAction__justified: "call-to-action-justified",
+        callToAction__disabled: "call-to-action-disabled",
+    };
+    const href: string = "#";
+
+    test("should implement unhandledProps", () => {
+        const handledProps: ICallToActionProps & ICallToActionManagedClasses = {
+            managedClasses,
+            href,
+            children: "text"
+        };
+
+        const unhandledProps: ICallToActionUnhandledProps = {
+            "data-my-custom-attribute": true
+        };
+
+        const props: CallToActionProps = {...handledProps, ...unhandledProps};
+
+        const rendered: any = shallow(
+            <Component {...props} />
+        );
+
+        expect(rendered.first().prop("data-my-custom-attribute")).toEqual(true);
+    });
+
+    // tslint:disable-next-line:max-line-length
+    test("should set a className that matches callToAction__primary managedClass when CallToActionAppearance.primary is passed to the appearance prop", () => {
```

Comment:
Try to avoid implementation details in the test descriptions:
"should apply a 'primary' html class when appearance is primary" is shorter and describes the *intent* of the test but not how the test should be implemented.

Can you apply this pattern to the other tests as well?

Reply:
All updated